### PR TITLE
Update CI configuration for building Redox libraries

### LINKED ISSUE #59254: Update toolchain for Redox dist
The old APT toolchain is no longer available

### PR DESCRIPTION
**Pull request by jackpot51**
This fixes https://github.com/rust-lang/rust/issues/59254

### CODE REVIEW

**Comment by sanxiyn** [comment]
```
./install-x86_64-redox.sh: line 6: wget: command not found
```

File: src/ci/docker/dist-various-1/install-x86_64-redox.sh [hunk 1]
```diff
@@ -3,11 +3,5 @@
...
 
-apt-get update
-apt-get install -y --no-install-recommends software-properties-common apt-transport-https
-
-apt-key adv --batch --yes --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys AA12E97F0881517F
-add-apt-repository -y 'deb https://static.redox-os.org/toolchain/apt /'
-
-apt-get update
-apt-get install -y x86-64-unknown-redox-gcc
+wget -O - https://static.redox-os.org/toolchain/x86_64-unknown-redox/relibc-install.tar.gz | \
```
Comment (kennytm) [inline src/ci/docker/dist-various-1/install-x86_64-redox.sh]:
```suggestion
curl https://static.redox-os.org/toolchain/x86_64-unknown-redox/relibc-install.tar.gz | \
```

Comment (kennytm) [reply src/ci/docker/dist-various-1/install-x86_64-redox.sh]: I've applied this manually.

**Comment by kennytm** [comment]
(The stable-nomination is relevant only if a point release is required)

File: src/ci/docker/dist-various-1/install-x86_64-redox.sh (hunk 1 above)
Comment (jackpot51) [reply src/ci/docker/dist-various-1/install-x86_64-redox.sh]: Thanks for that

**Pull request merged**